lib/enum/arrays.rb

```diff
@@ -1,24 +1,36 @@
 class Array
+  attr_accessor :array
+
+  def initialize(array)
+    @array = array
+  end
 
   def sum_of_numeric_elements
+    array.select {|elem| elem.is_a?(Numeric)}.inject {|sum, elem| sum + elem}
   end
 
   def product_of_numeric_elements
+    array.select {|elem| elem.is_a?(Numeric)}.inject {|product, elem| product * elem}
   end
 
   def even_numeric_elements
+    array.select {|elem| elem.is_a?(Fixnum)}.select {|elem| elem.even?}
   end
 
   def odd_numeric_elements
+    array.select {|elem| elem.is_a?(Fixnum)}.select {|elem| elem.odd?}
```

Comment:
Same here and for the next several problems - you can combine the two select statements